src/router/storeSync/LayerParamConfig.class.js

```diff
@@ -43,51 +42,49 @@ export function transformLayerIntoUrlString(layer, defaultLayerConfig) {
  * @returns {KMLLayer | ExternalWMTSLayer | ExternalWMSLayer | null} Will return an instance of the
  *   corresponding layer if the given layer is an external one, otherwise returns `null`
  */
-export function transformParsedExternalLayerIntoObject(parsedLayer) {
+export function createLayerObject(parsedLayer) {
+    let layer = parsedLayer
     // format is :  KML|FILE_URL|LAYER_NAME
     if (parsedLayer.id.startsWith('KML|') && parsedLayer.id.split('|').length === 3) {
         const splitLayerId = parsedLayer.id.split('|')
-        return new KMLLayer(
-            parsedLayer.opacity,
+        layer = new KMLLayer(
+            splitLayerId[1], // kml url
             parsedLayer.visible,
-            splitLayerId[1],
-            null,
+            parsedLayer.opacity,
+            null, // fileId, null := parsed from url
```

Comment:
some python operator in the comments, now you've shown you true colours 🙈

Reply:
@pakb nope it is not python `:=` doesn't exist neither in python. But its a mathematical notation for `defined to/as`  meaning that `null` is defined as parsed url.